card/lib/card/format.rb

```diff
@@ -54,6 +54,7 @@ def initialize card, opts={}
       opts.each { |key, value| instance_variable_set "@#{key}", value }
 
       @mode ||= :normal
+      @native_mode ||= @mode
```

Comment:
In general, I'm uncomfortable that such narrow solutions are coming up in such general places.  Here we're introducing @native_mode in the initialization method of, arguably, our second most important class.  (in MoFoS, Card = Mo and Card::Format = Fo.  Maybe it's a tie for second with Card::Set, which = S).

@root and @depth are very natural things to be here, and we could (and I should) explain them with something like: Formats are hierarchical, and any given format can have a parent and/or children.  @root is the Format object at the top of the tree, and the depth refers to the generational depth of this object.

@mode should have a similar explanation.

@native_mode seems so much more complex, no?